zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/RoleControllerTest.java

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.gateway.rest.controller;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import io.camunda.identity.rolemanagement.model.Role;
+import io.camunda.identity.rolemanagement.service.RoleService;
+import io.camunda.zeebe.gateway.protocol.rest.RoleRequest;
+import io.camunda.zeebe.gateway.protocol.rest.RoleResponse;
+import io.camunda.zeebe.gateway.protocol.rest.RoleSearchResponse;
+import io.camunda.zeebe.gateway.protocol.rest.SearchQueryRequest;
+import io.camunda.zeebe.gateway.rest.controller.usermanagement.RoleController;
+import java.net.URI;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ProblemDetail;
+
+@WebMvcTest(RoleController.class)
+public class RoleControllerTest extends RestControllerTest {
+
+  @MockBean private RoleService roleService;
+
+  @Test
+  void getRoleByIdShouldReturnExistingRole() {
+    final Role role = new Role();
+    role.setName("test");
+
+    final RoleResponse roleResponse = new RoleResponse();
+    roleResponse.setName("test");
+
+    when(roleService.findRoleByName("test")).thenReturn(role);
+
+    webClient
+        .get()
+        .uri("/v2/roles/test")
+        .accept(MediaType.APPLICATION_JSON)
+        .exchange()
+        .expectStatus()
+        .is2xxSuccessful()
+        .expectBody(RoleResponse.class)
+        .isEqualTo(roleResponse);
+  }
+
+  @Test
+  void getRoleByIdThrowsExceptionWhenServiceThrowsException() {
```

Comment:
🔧 You could test the different 400 and 500 status codes for different exceptions.